### PR TITLE
PD-554 - widget bus logic and validation

### PR DESCRIPTION
**Pull request by elisechant**

### CODE REVIEW

File: app/views/portal/index.html.erb [hunk 1]
```diff
@@ -24,7 +24,29 @@
       lastName: '',
       canEditDashboard: [1, 2]
     },
-    app: {},
+    app: {
+      SELECT_WIDGET_TYPE: [
+        { value: 'full', label: 'Full' },
+        { value: 'kpi-sparkline', label: 'Kpi Sparkline' },
```
Comment (joahua) [inline app/views/portal/index.html.erb]: "KPI Sparkline" probably correcter-er?

Comment (elisechant) [inline app/views/portal/index.html.erb]: This is a bigger task which is what the humanised labels should be. Clearly "full" is also not idea.

Comment (joahua) [inline app/views/portal/index.html.erb]: True. Somehow totally missed full. Never go full widget!

**Comment by elisechant** [comment]
Branch from feature/PD-553. Do not merge before https://github.com/AusDTO/dto-dashboard/pull/171

**Comment by elisechant** [comment]
Incorporated by merge: https://github.com/AusDTO/dto-dashboard/pull/177

**Pull request closed without merging**